pages/index.tsx

```diff
@@ -25,20 +25,22 @@ const books = [
 ];
 
 export default function Home() {
-  const [showLeftButton, setShowLeftButton] = useState(false);
+  const [showLeftButton, setShowLeftButton] = useState(true);
   const [showRightButton, setShowRightButton] = useState(true);
+  let [scroll, setScroll] = useState(0);
+  const snapContainer = useRef(null);
 
   useEffect(() => {
-    const snapContainer = document.querySelector("#snap-container");
-    if (snapContainer.scrollLeft === 0) {
+    if (scroll < 10) {
       setShowLeftButton(false);
-      // } else if (snapContainer.scrollLeft === snapContainer.scrollLeftMax) {
-      //   setShowRightButton(false);
+    } else if (scroll > snapContainer.current.scrollWidth - 600) {
+      setShowRightButton(false);
     } else {
       setShowLeftButton(true);
```

Comment:
this might do too many rerenders. You could only do this when showLeftButton === false